server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -323,7 +324,15 @@ public SearchService(
             settings,
             circuitBreakerService.getBreaker(CircuitBreaker.REQUEST)
         );
-        this.indexSearcherExecutor = indexSearcherExecutor;
+        // use the index searcher executor provided by plugin or the default one in core
+        if (indexSearcherExecutor != null) {
```

Comment:
Please revert this change, it goes back to https://github.com/opensearch-project/OpenSearch/pull/7203/files#r1171283310, the decision what to use should be taken in single place by `SearchModule`: check plugins -> check feature flag -> use one or none.